### ISSUE TITLE
Joining tables results in some null properties

### ISSUE THREAD

**Issue by tobrunvannuland** [Raizlabs/DBFlow#67]
Left join between 2 tables:
```
   final List<ProductRow> rows = new Select()
                .from(ProductRow.class)
                .join(PriceRecordRow.class, Join.JoinType.LEFT)
                .using(ProductRow$Table.ID)
                .queryList();
```
Returns a List of ProductRows, 2 fields of those items are not correctly initialised with values 
(amount & unit fields not correctly initialised, tried both String as int representation).

Without a Join:
```
   final List<ProductRow> rows = new Select()
                .from(ProductRow.class)
                .queryList();
```
Returns a List of ProductRows, all fields are correctly initialised with values.

**Comment by agrosner** [comment]
Thanks for your issue report. I am working on getting a feature in that
will enable you to define ```@Join``` model classes that enable merging of
columns. Right now its fairly limited. Hopefully by the end of the week it
will be in.

On Wed, Feb 11, 2015 at 11:18 AM, Tobrun Van Nuland <
notifications@github.com> wrote:

> Left join between 2 tables:
>
>    final List<ProductRow> rows = new Select()
>                 .from(ProductRow.class)
>                 .join(PriceRecordRow.class, Join.JoinType.LEFT)
>                 .using(ProductRow$Table.ID)
>                 .queryList();
>
> Returns a List of ProductRows, 2 fields of those items are not correctly
> initialised with values
> (amount & unit fields not correctly initialised, tried both String as int
> representation).
>
> Without a Join:
>
>    final List<ProductRow> rows = new Select()
>                 .from(ProductRow.class)
>                 .queryList();
>
> Returns a List of ProductRows, all fields are correctly initialised with
> values.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Raizlabs/DBFlow/issues/67>.
>



-- 
--

*Andrew Grosner, Android Developer*

Mobile Strategy, Design & Development

www.raizlabs.com | Check my calendar
<https://www.google.com/calendar/embed?src=andrew.grosner@raizlabs.com&ctz=America/New_York&mode=WEEK>
 | @agrosner <http://www.twitter.com/agrosner> | 716-472-5075
[image: Raizlabs]

**Comment by tobrunvannuland** [comment]
I'm not sure that this will resolve the problem, the referenced fields only occur in the ProductRow so I believe it is not linked too the joining mechanism itself (but is a small side effect).  If I find some time, I will add sources and do some debugging.

Thanks for the quick reply,

**Comment by agrosner** [comment]
What is the SQL statement that is generated @tobrunvannuland ? thanks. It may be with how its generated or used. Thank you!

**Comment by tobrunvannuland** [comment]
After investigating the query and trying to do manipulations as a rawQuery. I can only conclude that this isn't a bug from dbflow.

Thanks again for this awesome fast library.

**Comment by tsuharesu** [comment]
I'm having quite the same problem. The query is this:
```
SELECT * FROM `Thing' AS T INNER JOIN `User` AS U ON `T`.`authorName`=U.name WHERE `U`.`id`=1 ORDER BY D.id 
```
It is returning the Things correctly, but all ids (I have a column named id) are 1.

**Comment by tsuharesu** [comment]
I tried doing `Select("T.*")` but it don't work too, DBFlow is putting backticks on *

**Comment by faifai21** [comment]
I think I just hit this bug but in an odd way. I was using an inner join, but the models I got back had the id's of the corresponding rows in the right table. This was my WhereClause:
`new Select().from(Client.class).as("Client").join(ApptPreview.class, Join.JoinType.INNER)
                .on(Condition.column("Client." + Client$Table.ID).eq(ApptPreview$Table.CLIENTID)).where(
                        Condition.column(ApptPreview$Table.BEGINAT).greaterThan(new Date())).groupBy(
                        new QueryBuilder().appendQuoted("Client." + Client$Table.ID));`

I think the bug had something to do with the fact that both tables had the same column for ID ("id"). The bug still surfaced when I used aliases. I managed to fix this bug by making the column names unique (I changed "id" on the Client table to "cltid". This isn't really the best course of action, but I guess it isn't too bad since we can just use migrations to update the table on existing databases.
